### DISCUSSION TITLE
How to revert to previous version of KQCircuits?

### DISCUSSION THREAD

**Discussion by GuyMoshel** [iqm-finland/KQCircuits#27] [Q&A]
In KLayout package manager I can only install the newest version of KQCircuits (currently 4.5.0). Is there a way to revert back to version 4.4.0?

**Comment by nikosavola** [comment]
For handling versions, the 'developer setup' is recommended, see the recent discussion in https://github.com/iqm-finland/KQCircuits/issues/25#issuecomment-1323243113. With that setup you can simply use git the version control and rollback the repository to a tag (version) found in https://github.com/iqm-finland/KQCircuits/tags with for example
```bash
git checkout v4.4.31
```

**Comment by qpavsmi** [comment]
I also recommend the developer setup, but if you prefer to use Salt package, you can do the following:
1. Uninstall the current KQCircuits from package manager, close KLayout
2. Create a xml file with following content, replacing @KQC_VERSION@ with a KQC version of your choice:
```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Delivered by Salt.Mine version 1.22 (production) -->
<salt-mine xmlns="http://www.klayout.org/salt-mine/2017">
  <salt-grain>
    <name>KQCircuits</name>
    <version>@KQC_VERSION@</version>
    <api-version>0.26</api-version>
    <title>KQCircuits</title>
    <doc>KQCircuits is a KLayout/Python-based superconducting quantum circuit library developed by IQM.</doc>
    <doc-url>https://iqm-finland.github.io/KQCircuits/</doc-url>
    <url>https://github.com/iqm-finland/KQCircuits.git/tags/v@KQC_VERSION@/klayout_package</url>
    <license>GPLv3</license>
    <icon></icon>
  </salt-grain>
</salt-mine>
```
3. Set an environment variable in you OS so that KLAYOUT_SALT_MINE points to the xml file you created previously
4. Open KLayout and package manager. There should now only be KQCircuits available with a version of your choice
5. To revert the KLayout package manager to use the package list from the internet, unset the KLAYOUT_SALT_MINE environment variable.
